kube_operator/client.go

```diff
@@ -0,0 +1,361 @@
+package kube_operator
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	yaml "gopkg.in/yaml.v2"
+	"io"
+	"io/ioutil"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	apiv1beta1client "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	dynamic "k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	rest "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"reflect"
+	"strings"
+	"time"
+)
+
+const (
+	// TEST NAMESPACE
+	ANAX_NAMESPACE = "ibm-edge"
+
+	K8S_ROLE_TYPE           = "Role"
+	K8S_ROLEBINDING_TYPE    = "RoleBinding"
+	K8S_DEPLOYMENT_TYPE     = "Deployment"
+	K8S_SERVICEACCOUNT_TYPE = "ServiceAccount"
+	K8S_CRD_TYPE            = "CustomResourceDefinition"
+)
+
+type YamlFile struct {
+	Header tar.Header
+	Body   string
+}
+
+type APIObjects struct {
+	Type   *schema.GroupVersionKind
+	Object interface{}
+}
+
+type KubeClient struct {
+	Client *kubernetes.Clientset
+}
+
+func NewKubeClient() KubeClient {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		panic(err.Error())
```

Comment:
The use of panic should be avoided during runtime is possible. We try to limit its use to start up and initialization of workers. I see that you havent hooked in the worker to call this client. If the worker called these NewXXX methods during the constructor or during start, then panic is probably ok.